pygments/lexers/typst.py

```diff
@@ -81,24 +113,47 @@ class TypstLexer(RegexLexer):
             (r'\(|\{', Punctuation, 'code'),
             (r'\)|\}', Punctuation, '#pop'),
             (r'"[^"]*"', String.Double),
-            (r'[=,]', Operator),
+            (r',|\.{1,2}', Punctuation),
+            (r'=', Operator),
             (words(('and', 'or', 'not'), suffix=r'\b'), Operator.Word),
             (r'=>|<=|==|!=|>|<|-=|\+=|\*=|/=|\+|-|\\|\*', Operator), # comparisons
-            (r'([a-zA-Z_][a-zA-Z0-9_]*)(:)', bygroups(Name.Variable, Punctuation), '#push'),
-            (r'([a-zA-Z_][a-zA-Z0-9_]*)(\()', bygroups(Name.Function, Punctuation), '#push'),
+            (r'([a-zA-Z_][a-zA-Z0-9_-]*)(:)', bygroups(Name.Variable, Punctuation)),
+            (r'([a-zA-Z_][a-zA-Z0-9_-]*)(\()', bygroups(Name.Function, Punctuation), 'code'),
             (words(('as', 'break', 'export', 'continue', 'else', 'for', 'if',
-                    'import', 'in', 'include', 'return', 'while'), suffix=r'\b'),
+                    'in', 'return', 'while'), suffix=r'\b'),
              Keyword.Reserved),
+             (words(('import', 'include'), suffix=r'\b'), Keyword.Namespace),
             (words(('auto', 'none', 'true', 'false'), suffix=r'\b'), Keyword.Constant),
             (r'([0-9.]+)(mm|pt|cm|in|em|fr|%)', bygroups(Number, Keyword.Reserved)),
+            (r'0x[0-9a-fA-F]+', Number.Hex),
+            (r'0b[01]+', Number.Binary),
```

Comment:
Should be `Bin` according to https://github.com/pygments/pygments/blob/f64833d9d0443ff2443f8f8a4357c00196734d6c/pygments/token.py#L181

```suggestion
            (r'#0b[01]+', Number.Bin),
```